### DISCUSSION TITLE
Logi6502

### DISCUSSION THREAD

**Discussion by Eddieghoul** [logisim-evolution/logisim-evolution#1362] [Q&A]
I found a library called Logi6502.jar.  It works fine with the original version of Logisim, but with Logisim-evolution, it has an issue and aborts the simulation as soon as you drop the component on the screen.  Is there a way to fix this so that it works properly?

**Comment by R3dst0ne** [comment]
Could you please provide a link to it?

**Reply by Eddieghoul** [reply]
He has a Github page for it.  Since I posted this, I recently found that someone had made up another 6502 to work on logisim-evolution.  At the moment, I'm using it to recreate the demo circuit that comes with logi6502 for the CBurch version of Logisim so I can at least give it a little bit of a workout.  The Logisim-Evolution version also has a Github page.

**Comment by Emil-DV** [comment]
Is it possibly this one? https://github.com/SolraBizna/logi6502

I'm going to see if I can get this working under Logisim-Evolution (looks like it was designed for OG logisim).  It looks like a great starting point for my own processor lib.

**Reply by Eddieghoul** [reply]
Yes, that's the one.

**Comment by Emil-DV** [comment]
I've got this working under [Logisim-Evolution v3.5+](https://github.com/Emil-DV/logi6502/tree/LogisimEvolution35%2B)

**Reply by Eddieghoul** [reply]
OUTSTANDING!!!  I'm going to play with it this weekend.  Thank you!

**Reply by Eddieghoul** [reply]
You wouldn't happen to have a link to some better documentation for logisim-evolution.  All I can find is the intro stuff that was copy and pasted from OG logisim.  I'm specifically interested in the SOC library.

**Reply by Emil-DV** [reply]
Unfortunately I don't have anything more than what's on this github location.  You might try searching youtube for a project that was done with the SOC components.

**Reply by BFH-ktt1** [reply]
This is my "lack of time". There is documentation on the SOC library (not complete), which is a bit "hidden". In the help system go to "Library Reference" main page, and you find in the list "System On Chip components library". Here you find the current available documentation. I'm sorry for the inconvenience, it's still on my TODO list

**Reply by Eddieghoul** [reply]
Thanks, guys.  I appreciate the help.

**Reply by Eddieghoul** [reply]
I am so not a Java programmer.  Do you happen to have the JAR file already compiled?  Or is it in one of the folders and I just didn't see it?

**Reply by Emil-DV** [reply]
https://github.com/Emil-DV/logi6502/releases

**Reply by Eddieghoul** [reply]
Thanks again.
